src/index.ml

```diff
@@ -80,6 +80,40 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
 
   let entry_sizeL = Int64.of_int entry_size
 
+  module Fan = struct
+    type t = { size : int; fans : int64 array; mask : int; shift : int }
+
+    let v n =
+      let size = n in
+      let nb_fans = 1 lsl size in
+      let fans = Array.make nb_fans (-1L) in
+      let shift = K.hash_size - size in
```

Comment:
You could take `hash_size` as an argument so you don't depend on anything. `shift` is stored in `t` so this is free.